### PR TITLE
Copter: move mavlink set-position defines into mavlink library

### PR DESCRIPTION
**Pull request by peterbarker**
These are mavlink-specific defines, so type them and put them in the function that uses them.

### CODE REVIEW

File: ArduCopter/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -880,6 +880,14 @@ void GCS_MAVLINK_Copter::handle_mount_message(const mavlink_message_t &msg)
 
 void GCS_MAVLINK_Copter::handleMessage(const mavlink_message_t &msg)
 {
+    // for mavlink SET_POSITION_TARGET messages
+    constexpr uint32_t MAVLINK_SET_POS_TYPE_MASK_POS_IGNORE      = ((1<<0) | (1<<1) | (1<<2));
```
Comment (khancyr) [inline ArduCopter/GCS_Mavlink.cpp:884]: maybe we can use https://mavlink.io/en/messages/common.html#POSITION_TARGET_TYPEMASK

**Comment by khancyr** [comment]
@peterbarker ping on this one ! easy to merge with or without the fix !

**Comment by khancyr** [comment]
In fact, NO ... this need to be better as all vehicles can benefit of those define. Now that we have mavlink enum for this, we should use the MAVLINK stuff on all vehicle and remove those defines from copter,rover and sub

File: ArduCopter/GCS_Mavlink.cpp (hunk 1 above)
Comment (peterbarker) [reply ArduCopter/GCS_Mavlink.cpp]: Done

**Comment by khancyr** [comment]
@peterbarker could you move this either into AP_Vehicle or gsc_mavlink ? all vehicles need those. For example, Rover and Sub got the same define as copter in defines.h
That would be a nice clean up !

**Comment by peterbarker** [comment]
> @peterbarker could you move this either into AP_Vehicle or gsc_mavlink ? all vehicles need those. For example, Rover and Sub got the same define as copter in defines.h
> That would be a nice clean up !

No, they don't have the same defines.  I checked before creating the PR :-)

Sub *probably* is identical - I didn't check - but Rover is definitely not.

Given the fundamentally different movement modes of the vehicles (excepting Sub/Copter), I don't think attempting to share here would gain us anything but pain.

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**